src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -384,8 +385,13 @@ public void validateSQLSyntax(FailureCollector failureCollector, BigQuery bigQue
       try {
         bigQuery.create(JobInfo.of(queryJobConfiguration));
       } catch (BigQueryException e) {
-        failureCollector.addFailure(String.format("%s.", e.getMessage()), "Please specify a valid query.")
-          .withConfigProperty(SQL);
+        if (e.getCode() == ERROR_CODE_NOT_FOUND)  {
+          LOG.warn("The query contains a non resource that doesn't exist. Error code {} and message generated {}",
```

Comment:
@sanjanasandeep is this the intended behaviour?

Reply:
I thought about this some more. I think the problem is that this specific situation (that the bug was created for) occurs when there is a dependency between the action and the preceding source.

However, the pipeline can also contain only an action plugin (or a set of action plugins). At that time, it may be desirable to fail if a table is provided that does not exist?

Unless the behavior we want to shoot for is that in general we don't want to check for table existence at deployment time, and rely upon it being a runtime error.

Ideally, this would be one of those scenarios where we would extend the failureCollector/validation APIs to also support warnings, and not just errors like it does now.


But until then, I think we may just want to leave it as-is, because it could have unintended consequences. E.g. with this change, if I mis-typed the name of the table in my query, then it won't fail until the pipeline is actually run. This will force the user to then duplicate the pipeline, fix the error and try again. As opposed to that, it might be better to fail early, let the user know, and if they want to ignore the failure, then use a macro.

Reply:
Thinking a little more, I think the right thing to do will be, keep the validation, but change the error message. So, if the error from BQ is that the table is not found, update the messagee in the failure collector to be something to the effect of "Table was not found. Please verify the table name. If the table will be created at runtime, update to use a macro for the table name"